inductiva/simulators/fvcom.py

```diff
@@ -0,0 +1,93 @@
+        """Initialize the FVCOM simulator.
+        
+        Args:
+            version (str): The version of the simulator to use. If None, the
+                latest available version in the platform is used.
+            use_dev (bool): Request use of the development version of
+                the simulator. By default (False), the production version
+                is used.
+        """
+        super().__init__(version=version, use_dev=use_dev)
+        self.api_method_name = "fvcom.fvcom.run_simulation"
+
+    def run(self,
+            input_dir: str,
+            debug: int = 0,
+            case_name: str = "",
+            compilation: str = "",
+            use_hwthread: bool = True,
+            create_namelist: str = "",
+            n_vcpus: Optional[int] = None,
+            working_dir: Optional[str] = "",
+            storage_dir: Optional[str] = "",
+            resubmit_on_preemption: bool = False,
+            extra_metadata: Optional[dict] = None,
+            on: Optional[types.ComputationalResources] = None,
+            **kwargs) -> tasks.Task:
+        """Run the simulation.
+
+        Args:
+            input_dir: Path to the directory of the simulation input files.
+            casename: Name of the simulation case.
+
+            debug: Debug level of the simulation (from 0 to 7).
+
+            compilation: At the current moment we provide users with two 
+            options:
+                - None (default): Uses default fvcom binary.
+                - 'estuary': Uses the fvcom_estuary binary.
+                The flags used to compile each binary can be found in the docker
+                image at /make.inc and /make_estuary.inc.
+            
+            create_namelist: Used to create a namelist file for the simulation.
+                Example: 'create_namelist=hello' will create hello_run.nml in
+                    the working_dir.
+
+            working_dir: Path (relative to the input directory) to the directory
+                where the simulation nml file is located. If not provided, the
+                input directory is used.
+
+            n_vcpus: Number of vCPUs to use in the simulation. If not provided
+            (default), all vCPUs will be used.
+
+            use_hwthread: If specified Open MPI will attempt to discover the
+            number of hardware threads on the node, and use that as the
+            number of slots available.
+
+            on: The computational resource to launch the simulation on. If None
+                the simulation is submitted to a machine in the default pool.
+
+            resubmit_on_preemption (bool): Resubmit task for execution when
+                previous execution attempts were preempted. Only applicable when
+                using a preemptible resource, i.e., resource instantiates with
+                `spot=True`.
+
+            other arguments: See the documentation of the base class.
+        """
+
```

Comment:
please add a validation to check that the `compilation` argument is either `None` or `"estuary"`.
Also, I'm not sure I like the name `"compilation"`. They are actually "models". what do you think of renaming it to models?